boards/riscv/tlsr9258a/tlsr9258a_defconfig

```diff
@@ -11,3 +11,4 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_SERIAL=y
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
+CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE=y
```

Comment:
with 2 wire you will fail some of certification tests
As well OTA will be not stable

Reply:
why we change the spi to 2 wire ,because after poweroff , it will have rate to revert back 2 wire , so we need to start with 2 -wire , and then detect the status in  soc_b9x_check_flash to set back to 4 wire . 
so we have two way to slove this problem . 
1. set the config of the     CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE=y
2. change the cstart settings , just set the 4-wire option to 2 wire value . 
just like that :
                                                              #if defined(CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE) && CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE
	.word (0x030020A8)              #/ 2 - wire SPI interface
#else
	.word (0x030020A8)              #/ 4 - wire SPI interface will set in the soc_b9x_check_flash
#endif                     
 
        And I aggree your opinion ,  don't set this define .
        But if we want the define  CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE  take effect ,should add the define to disable the 4-wire func init in the func of 
                                                                       /* API implementation: driver initialization */
static int flash_b9x_init(const struct device *dev)
{
	struct flash_b9x_data *dev_data = dev->data;

	k_mutex_init(&dev_data->flash_lock);

	flash_change_rw_func(flash_4read, flash_quad_page_program); // this func should add special proc to exclude if the 2-wire take effect .

	return 0;
}

Reply:
something like that to control to force use 2-wire 
#if !CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE
flash_change_rw_func(flash_4read, flash_quad_page_program);
#endif

Reply:
ok now its clear for me.
Anyway option to use "CONFIG_TELINK_B9X_2_WIRE_SPI_ENABLE=y" is not correct.
When we set 2 wire - we have to use 2 wire only.
I think option 2 is mach clear for understanding and just leave comment for 4 wire like you shown
.word (0x030020A8) #/ 4 - ....
And in place where you set real 4 wire make it for B95 only in flash driver.